### PR TITLE
docs: Add new page to explain how to deploy Teleirc

### PR DESCRIPTION
**Pull request by jwflory**
### Blocked by PR #121 – review it first!

---

This commit is part of issue #118.

A new page is added specifically to explain how to deploy Teleirc. It
consolidates the **Usage** section from `installation.rst` and all of
the `using-docker.rst` page. This commit intends to make deployment
options more clear and easy to understand.

Some methods were removed, like the ArchLinux package. It is many
updates behind on the bot, so I think it's better to recommend people to
use directly from source where possible.

Additionally, I refactored the installation guide from changes in #120, #121, and #122.

Screenshot preview of "How to deploy Teleirc" page

### CODE REVIEW

**Comment by thenaterhood** [comment]
Ayyy, I forgot I maintained the Arch package. I updated it, but I also created a "direct-from-git" one which is probably more what we're looking for if we're still interested in having an Arch package - https://aur.archlinux.org/packages/teleirc-git/

If not, all good, no need to officially support it here.

File: docs/deploy-teleirc.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+This page offers suggestions on possible deployment options.
+
+
+*******
+systemd
+*******
+
+The **recommended deployment method** is with `systemd <https://en.wikipedia.org/wiki/Systemd>`_.
+This method requires a basic understanding of systemd unit files.
+Create a unique systemd service for each Teleirc instance.
```
Comment (thenaterhood) [inline docs/deploy-teleirc.rst]: Inspired https://github.com/RITlug/teleirc/issues/123

Comment (jwflory) [reply docs/deploy-teleirc.rst]: @thenaterhood Thanks for pointing it out, I learned something new here. :+1:

**Comment by jwflory** [comment]
@thenaterhood Thanks for pointing out the git source package. I pushed commit 3a5a0134c8735eb9265b246feba3d455652596bb to add Arch Linux AUR instructions as a deployment option. I'm going to merge this PR in. :clapper:

**Pull request merged**